controllers/suggestions/suggest_apiwrite_review_controller.py

```diff
@@ -0,0 +1,145 @@
+
+        self.template_values.update({
+            'success': self.request.get("success"),
+            'suggestions': suggestions,
+            'auth_names': AuthType.type_names,
+        })
+        self.response.out.write(
+            jinja2_engine.render('suggest_apiwrite_review_list.html', self.template_values))
+
+    @ndb.transactional(xg=True)
+    def _process_accepted(self, suggestion_id, message):
+        suggestion = Suggestion.get_by_id(suggestion_id)
+        event_key = suggestion.contents['event_key']
+        user = suggestion.author.get()
+        event = Event.get_by_id(event_key)
+
+        auth_id = ''.join(
+            random.choice(string.ascii_lowercase + string.ascii_uppercase + string.digits) for _ in
+            range(16))
+        auth_types = self.request.get_all("auth_types", [])
+        expiration_offset = int(self.request.get("expiration_days"))
+        if expiration_offset != -1:
+            expiration_event_end = event.end_date + timedelta(days=expiration_offset + 1)
+            expiration_now = datetime.now() + timedelta(days=expiration_offset)
+            expiration = max(expiration_event_end, expiration_now)
+        else:
+            expiration = None
+        auth = ApiAuthAccess(
+            id=auth_id,
+            description="{} @ {}".format(user.display_name, suggestion.contents['event_key']),
+            secret=''.join(
+                random.choice(string.ascii_lowercase + string.ascii_uppercase + string.digits) for _
+                in range(64)),
+            event_list=[ndb.Key(Event, event_key)],
+            auth_types_enum=[int(type) for type in auth_types],
+            owner=suggestion.author,
+            expiration=expiration
+        )
+        auth.put()
+
+        suggestion.review_state = Suggestion.REVIEW_ACCEPTED
+        suggestion.reviewer = self.user_bundle.account.key
+        suggestion.reviewed_at = datetime.now()
+        suggestion.put()
+
+        return user, event_key, """Hi {},
+
+We have approved your request for auth tokens so you can add data to the following event: {} {}
+
+You can find the keys on your account overview page: https://www.thebluealliance.com/account
+{}
+If you have any questions, please don't heasitate to reach out to us at contact@thebluealliance.com
+
+Thanks,
+TBA Admins
+            """.format(user.display_name, event.year, event.name, message)
+
+    def post(self):
+        self.verify_permissions()
+        suggestion_id = int(self.request.get("suggestion_id"))
+        verdict = self.request.get("verdict")
+        message = self.request.get("user_message")
+
+        email_body = None
+        user = None
+        event_key = None
+        status = ''
+        if verdict == "accept":
+            status = 'accept'
+            user, event_key, email_body = self._process_accepted(suggestion_id, message)
+
+        elif verdict == "reject":
+            suggestion = Suggestion.get_by_id(suggestion_id)
```

Comment:
Nevermind, there's only one DB write for reject, and if the transaction fails it will throw an error so no email will go out.